app/src/main/java/fr/free/nrw/commons/wikidata/WikiBaseClient.java

```diff
@@ -0,0 +1,45 @@
+package fr.free.nrw.commons.wikidata;
+
+import org.wikipedia.csrf.CsrfTokenClient;
+import org.wikipedia.dataclient.mwapi.MwQueryResponse;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import fr.free.nrw.commons.upload.WikiBaseInterface;
+import io.reactivex.Observable;
+
+import static fr.free.nrw.commons.di.NetworkingModule.NAMED_WIKI_DATA_CSRF;
+
+/**
+ * Wikibase Client for calling WikiBase APIs
+ */
+@Singleton
+public class WikiBaseClient {
+
+    private final WikiBaseInterface wikiBaseInterface;
+    private final CsrfTokenClient csrfTokenClient;
+
+    @Inject
+    public WikiBaseClient(WikiBaseInterface wikiBaseInterface,
+                          @Named(NAMED_WIKI_DATA_CSRF) CsrfTokenClient csrfTokenClient) {
+        this.wikiBaseInterface = wikiBaseInterface;
+        this.csrfTokenClient = csrfTokenClient;
+    }
+
+    public Observable<Boolean> postEditEntity(String fileEntityId, String data) {
```

Comment:
javadoc on methods

Reply:
@nicolas-raoul I hahve not much sure about this class, it wasn't added by me it was a result of merge with VitalyPinchuk's branch